script/swap_and_bridge/contracts/SwapAndBridge_e2e_test.s.sol

```diff
@@ -0,0 +1,324 @@
+            "SentMessage: Invalid gas limit, not matching contract MIN_DEPOSIT_GAS"
+        );
+
+        // The message is encoded in a weird way: bytes 4 is packed, the addresses are not
+        // Hence, we slice the message to remove the bytes4 selector.
+        require(
+            bytes4(getSlice(1, 5, message))
+                == bytes4(keccak256("finalizeBridgeERC20(address,address,address,address,uint256,bytes)")),
+            "SentMessage: Invalid selector"
+        );
+
+        (address localToken, address remoteToken, address from, address to, uint256 amount, bytes memory extraData) =
+            abi.decode(getSlice(5, message.length, message), (address, address, address, address, uint256, bytes));
+
+        require(remoteToken == _l1Token, "SentMessage: Invalid L1 token address");
+        require(localToken == _l2Token, "SentMessage: Invalid L2 token address");
+        require(from == address(swapAndBridge), "SentMessage: Invalid sender address");
+        require(to == testAccount, "SentMessage: Invalid recipient address");
+        require(amount == TEST_AMOUNT, "SentMessage: Invalid amount");
+        require(extraData.length == 2, "SentMessage: Invalid extra data");
+
+        vm.serializeAddress("", "sender", sender);
+        vm.serializeAddress("", "target", _l2Bridge);
+        vm.serializeBytes("", "message", message);
+        vm.serializeUint("", "messageNonce", messageNonce);
+
+        bytes memory data = abi.encode(sender, _l2Bridge, message, messageNonce, gasLimit);
+
+        string memory dataPath = string.concat(vm.projectRoot(), "/script/swap_and_bridge/example/message.data");
+        vm.writeFileBinary(dataPath, data);
+        console2.log("Transfer completed. Data saved to", dataPath);
+    }
+}
+
+/// @title TestE2EL2Script
+/// @notice This contract is used to test bridging tokens from L1 to L2 network.
+///         This contract runs the L2 part of it, by relaying the message that was emitted on the L1.
+contract TestE2EL2Script is Script {
+    // The L2 crossi domain messenger contract
+    IL2CrossDomainMessenger l2Messenger;
+
+    // Address used for E2E tests
+    address testAccount;
+
+    // The test value to be bridged
+    uint256 constant TEST_AMOUNT = 500 ether;
+
+    // L2 Cross Domain Messenger address
+    address constant L2_CROSS_DOMAIN_MESSENGER_ADDR = 0x4200000000000000000000000000000000000007;
+
+    // L2 sequencer address (this is the Lisk Sepolia Sequencer address)
+    address constant SEQUENCER_ADDR = 0x968924E6234f7733eCA4E9a76804fD1afA1a4B3D;
+
+    function setUp() public {
+        l2Messenger = IL2CrossDomainMessenger(L2_CROSS_DOMAIN_MESSENGER_ADDR);
+        testAccount = address(0xc0ffee);
+    }
+
+    function run(address _l2Token) public {
+        assert(_l2Token != address(0));
+        IWrappedETH l2LSTToken = IWrappedETH(payable(_l2Token));
+        string memory dataPath = string.concat(vm.projectRoot(), "/script/swap_and_bridge/example/message.data");
```

Comment:
Should `message.data` file be deleted at the end of test like it's done in below link?
https://github.com/LiskHQ/lisk-contracts/blob/development/test/utils/Utils.t.sol#L31

Reply:
I do delete the file using `vm.removeFile(dataPath);` on line 310

Reply:
You're right!
I didn't see it.